solutions/Abhilasha06/Assignment1/my_parser.py

```
@@ -0,0 +1,94 @@
+            mssg = "Error: --key can't occur more than once"
+        elif size != 2:
+            mssg = "Error: --key can't be empty"
+        elif bool(re.match('^[0-9]+$', value)):
+            data["--key"] = value
+            mssg = "successful"
+        else:
+            mssg = "Error: The value for the '--key' argument must be a positive integer."
+        return mssg
+    @classmethod
+    def check_local(cls, size, data):
+        """
+        to check if local is valid or not
+        """
+        if size != 1:
+            mssg1 = "Error: --local takes no value"
+        elif "--remote" in data:
+            mssg1 = "Error: --local and --remote can't occur together"
+        else:
+            data["--local"] = "true"
+            mssg1 = "successful"
+        return mssg1
+    @classmethod
+    def check_remote(cls, size, data):
+        """
+        to check if remote is valid or not
+        """
+        if size != 1:
+            mssg2 = "Error: --remote takes no value"
+        elif "--local" in data:
+            mssg2 = "Error: --local and --remote can't occur together"
+        else:
+            data["--remote"] = "true"
+            mssg2 = "successful"
+        return mssg2
+    @classmethod
+    def check_argument(cls, argument):
+        """
+        to parse the given input and give error if it is invalid
+        """
+        import json
+        data = {}
+        list_of_fields = ["--key", "--local", "--remote", "--name"]
+        argument = list(argument.split(" "))
+        no_of_args = (int)(format(len(argument)))
+        if no_of_args == 1:
+            ans = "Error: The '--key' argument is required, but missing from input."
+        for args in argument[1:]:
+            splitted_word = args.split('=')
+            size = int(len(splitted_word))
+            if size > 2:
+                ans = "Error: Invalid argument"
+            elif splitted_word[0] not in list_of_fields:
+                return"Error: Field not defined"
```

When you encounter cases like this, people usually raise exceptions, instead of returning.
That way, the caller of your library can assume that if it returned value, everything is fine, but if it didn't, then an exception would be raised that can be caught.